sentysis-angular/src/app/form/form.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormsModule, ReactiveFormsModule } from '@angular/forms';
+
+import { YtUrl } from './form-url';
+import { FormService } from './form.service';
+
+@Component({
+  selector: 'app-form',
+  providers: [FormService],
+  templateUrl: './form.component.html',
+  styleUrls: ['./form.component.css']
+})
+export class FormComponent implements OnInit {
+
+  ytUrls!: YtUrl[];
+  youtubeForm;
+  videoId = '';
+
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private formService: FormService
+  ) {
+    this.youtubeForm = this.formBuilder.group({
+      videoUrl: ''
+    });
+  }
+  ngOnInit() {}
+
+  onSubmit(newYtUrl) {
+    if(typeof newYtUrl.videoUrl == 'string'){
+      let stringYtUrl = newYtUrl.videoUrl;
+      if(this.verifyUrl(stringYtUrl)) {
+        console.log('GANGSTA');
+        this.formService
+        .addYtUrl(newYtUrl)
+        .subscribe(ytUrl => this.ytUrls.push(ytUrl));
+
+      }
+      else {
+        console.log('WRONG');
+      }
+      this.youtubeForm.reset();
+    }
+    this.youtubeForm.reset();
+  }
+
+  // helper methods
+
+  verifyUrl(ytUrl) {
```

Comment:
rename this to `isVerifyUrl`

Reply:
This also need to be move into its own function outside of the component so that you can test it.